src/main/java/org/radarcns/data/FileCache.java

```diff
@@ -90,6 +120,9 @@ public boolean writeRecord(GenericRecord record) throws IOException {
     public void close() throws IOException {
         recordConverter.close();
         writer.close();
+        if (tmpPath != null) {
+            Files.move(tmpPath, path, REPLACE_EXISTING);
+        }
```

Comment:
Will this also run if there is an unhandled exception ?? Like socket connection with hdfs is broken ?

Reply:
No, in that case the IOException will be propagated to the upper level, and the offsets file will not be updated. At the next iteration, it will retry.